plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/ui/LocalAppEngineServerWizardFragment.java

```diff
@@ -0,0 +1,103 @@
+
+import com.google.cloud.tools.appengine.api.AppEngineException;
+import com.google.cloud.tools.appengine.cloudsdk.CloudSdk;
+import com.google.cloud.tools.eclipse.appengine.localserver.Messages;
+import com.google.cloud.tools.eclipse.sdk.ui.preferences.CloudSdkPreferenceArea;
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.jface.preference.PreferenceDialog;
+import org.eclipse.osgi.util.NLS;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.ui.dialogs.PreferencesUtil;
+import org.eclipse.wst.server.ui.wizard.IWizardHandle;
+import org.eclipse.wst.server.ui.wizard.WizardFragment;
+
+public class LocalAppEngineServerWizardFragment extends WizardFragment {
+  final private boolean cloudSdkConfigured;
+  private String cloudSdkLocation = "";
+
+  public LocalAppEngineServerWizardFragment() {
+    cloudSdkConfigured = isCloudSdkConfigured();
+  }
+
+  @Override
+  public boolean hasComposite() {
+    return true;
+  }
+
+  @Override
+  public boolean isComplete() {
+    return cloudSdkConfigured;
+  }
+
+  @Override
+  public Composite createComposite(final Composite parent, IWizardHandle wizard) {
+    wizard.setTitle(Messages.CREATE_APP_ENGINE_RUNTIME_WIZARD_TITLE);
+    wizard.setDescription(Messages.CREATE_APP_ENGINE_RUNTIME_WIZARD_DESCRIPTION);
+    
+    Composite cloudSdkComposite = new Composite(parent, SWT.NONE);
+    GridLayout layout = new GridLayout();
+    layout.numColumns = 1;
+    cloudSdkComposite.setLayout(layout);
+  
+    Label label = new Label(cloudSdkComposite, SWT.NONE);
+    if (cloudSdkConfigured) {
+      label.setText(NLS.bind(Messages.RUNTIME_WIZARD_CLOUD_SDK_FOUND, cloudSdkLocation)); // TODO: not printing out correctly
+    } else {
+      label.setText(Messages.RUNTIME_WIZARD_CLOUD_SDK_NOT_FOUND);
+    }
+
+    Button button = new Button(cloudSdkComposite, SWT.PUSH);
+    button.setText("Update SDK Location");
+    button.addSelectionListener(new SelectionAdapter() {
+      @Override
+      public void widgetSelected(SelectionEvent event) {
+        Shell shell = parent.getShell();
+        if (!MessageDialog.openQuestion(shell, Messages.RUNTIME_WIZARD_OPEN_CLOUD_SDK_DIALOG_TITLE, 
+            Messages.RUNTIME_WIZARD_OPEN_CLOUD_SDK_DIALOG_MESSAGE)) {
+          return;
+        }
+        shell.close();
+
+        PreferenceDialog dialog =
+            PreferencesUtil.createPreferenceDialogOn(null, CloudSdkPreferenceArea.PAGE_ID, null, null);
+        dialog.open();
+      }
+    });
+    
+    return cloudSdkComposite;
+  }
+
+  private boolean isCloudSdkConfigured() {
+    try {
+      CloudSdk cloudSdk = new CloudSdk.Builder().build();
+      cloudSdkLocation = cloudSdk.getSdkPath().toString();
```

Comment:
perhaps this could return the location instead of setting the field, and return null when not configured